src/test/scala/cognite/spark/v1/AssetsHierarchyBuilderTest.scala

```diff
@@ -0,0 +1,152 @@
+package cognite.spark.v1
+
+import cognite.spark.v1.SparkSchemaHelper.fromRow
+import com.cognite.sdk.scala.v1.AssetCreate
+import org.apache.spark.sql.Row
+import org.scalatest.{FlatSpec, Matchers}
+
+class AssetsHierarchyBuilderTest  extends FlatSpec with Matchers with SparkTest {
```

Comment:
Add tests for an empty input, and an input where things aren't connected at all (lots of null/None values), and an input where there is a cycle.